src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -171,7 +172,12 @@ public Task<string> GetNormalizedUserNameAsync(IUser user, CancellationToken can
                 throw new ArgumentNullException(nameof(user));
             }
 
-            return Task.FromResult(((User)user).NormalizedUserName);
+            if (user is User su)
+            {
+                return Task.FromResult(su.NormalizedUserName);
```

Comment:
Okay, so here (and in other places) would need to replace `su` by another name.

Hmm, was it necessary to change all these tests and not use the casting anymore?
Here if user is not null, can it be something else than an `User`?

Reply:
Any suggestion on name?

Reply:
Not for now ;)

Or just use the old casting, or keep the UserStore unchanged, but as you want.

Hmm, it changes a little the behavior in some places by not throwing if it is not a `User`.

Otherwise I will let you know if I find a better name ;)

Also saw one in `NotifyUserTaskActivity`.

Reply:
The reason there is change to that file is because there was an actual change... and I made the syntax change then. Now, I don't recall of a change, but since the formatting is much cleaner, I rather keep it. I'll think of another name

Reply:
Maybe


            if (user is User usr)
            {
                usr.NormalizedUserName = normalizedName;
            }

Or why not just


            if (user is User u)
            {
                u.NormalizedUserName = normalizedName;
            }

Reply:
> Now, I don't recall of a change

Would be good to retrieve it to review this actual change, only if possible

Reply:
okay. I made all the changes around the naming.